tests/foreman/ui/test_classparameters.py

```diff
@@ -15,20 +15,132 @@
 
 :Upstream: No
 """
-
+from fauxfactory import gen_string
+from nailgun import entities
+from requests import HTTPError
+
+from robottelo.api.utils import (
+    create_role_permissions,
+    delete_puppet_class,
+    publish_puppet_module,
+)
+from robottelo.constants import CUSTOM_PUPPET_REPO
 from robottelo.decorators import (
     run_only_on,
     skip_if_bug_open,
     stubbed,
     tier1,
+    tier2,
     tier3
 )
+from robottelo.helpers import get_nailgun_config
 from robottelo.test import UITestCase
+from robottelo.ui.session import Session
 
 
 class SmartClassParametersTestCase(UITestCase):
     """Implements Smart Class Parameter tests in UI"""
 
+    @classmethod
+    def set_session_org(cls):
+        """Creates new organization to be used for current session the
+        session_user will login automatically with this org in context
+        """
+        cls.session_org = entities.Organization().create()
+
+    @classmethod
+    def setUpClass(cls):
+        """Import some parametrized puppet classes. This is required to make
+        sure that we have smart class parameter available.
+        Read all available smart class parameters for imported puppet class to
+        be able to work with unique entity for each specific test.
+        """
+        super(SmartClassParametersTestCase, cls).setUpClass()
+        cls.puppet_modules = [
+            {'author': 'robottelo', 'name': 'ui_test_classparameters'},
+        ]
+        cv = publish_puppet_module(
+            cls.puppet_modules, CUSTOM_PUPPET_REPO, cls.session_org)
+        cls.env = entities.Environment().search(
+            query={'search': u'content_view="{0}"'.format(cv.name)}
+        )[0].read()
+        cls.puppet_class = entities.PuppetClass().search(query={
+            'search': u'name = "{0}" and environment = "{1}"'.format(
+                cls.puppet_modules[0]['name'], cls.env.name)
+        })[0]
+        cls.sc_params_list = entities.SmartClassParameters().search(
+            query={
+                'search': 'puppetclass="{0}"'.format(cls.puppet_class.name),
+                'per_page': 1000
+            })
+
+    @classmethod
+    def tearDownClass(cls):
```

Comment:
Please prefer using cleanup funcs instead tearDownClass. Not a blocker for this particular case.